--- /etc/rc.subr	2014-01-16 22:41:39.000000000 +0000
+++ /etc/rc.subr	2014-08-30 18:11:13.000000000 +0000
@@ -464,6 +464,8 @@
 #
 #	${name}_fib	n	Routing table number to run ${command} with.
 #
+#	${name}_pmac	n	MAC label to run ${command} with.
+#
 #	${name}_nice	n	Nice level to run ${command} at.
 #
 #	${name}_user	n	User to run ${command} as, using su(1) if not
@@ -650,7 +652,7 @@
 	eval _chdir=\$${name}_chdir	_chroot=\$${name}_chroot \
 	    _nice=\$${name}_nice	_user=\$${name}_user \
 	    _group=\$${name}_group	_groups=\$${name}_groups \
-	    _fib=\$${name}_fib
+	    _fib=\$${name}_fib		_pmac=\$${name}_pmac
 
 	if [ -n "$_user" ]; then	# unset $_user if running as that user
 		if [ "$_user" = "$(eval $IDCMD)" ]; then
@@ -735,12 +737,14 @@
 				_doit="\
 ${_nice:+nice -n $_nice }\
 ${_fib:+setfib -F $_fib }\
+${_pmac:+setpmac $_pmac }\
 chroot ${_user:+-u $_user }${_group:+-g $_group }${_groups:+-G $_groups }\
 $_chroot $command $rc_flags $command_args"
 			else
 				_doit="\
 ${_chdir:+cd $_chdir && }\
 ${_fib:+setfib -F $_fib }\
+${_pmac:+setpmac $_pmac }\
 $command $rc_flags $command_args"
 				if [ -n "$_user" ]; then
 				    _doit="su -m $_user -c 'sh -c \"$_doit\"'"
